bioengine/engine.py

```diff
@@ -0,0 +1,128 @@
+"""The BioEngine class."""
+import inspect
+import sys
+import yaml
```

Comment:
We should use ruamel.yaml instead. It's better maintained and compliant to yaml spec.

Reply:
Let's save this for another PR.